testbench/common.py

```diff
@@ -671,6 +671,37 @@ def streamer():
     return response_handler
 
 
+def __get_streamer_response_custom_error_fn(
+    database, method, test_id, error_code, limit, chunk_size=4
+):
+    def response_handler(data):
+        def streamer():
+            d = _extract_data(data)
+            bytes_downloaded = 0
+            for r in range(0, len(d), chunk_size):
+                if bytes_downloaded < limit:
+                    chunk_end = min(r + chunk_size, len(d))
+                    chunk_downloaded = chunk_end - r
+                    bytes_downloaded += chunk_downloaded
+                    yield d[r:chunk_end]
+                if bytes_downloaded >= limit:
+                    database.dequeue_next_instruction(test_id, method)
+                    raise testbench.error.RestException(
+                        f"Retry Test: Caused a {error_code}. Fault injected after downloading {bytes_downloaded} bytes",
```

Comment:
For a large buffer, and probably only when using a real TCP/IP connection, the error code here has no effect on the client-side, just FYI.